### ISSUE TITLE
Implement Lidar data cost function 

### ISSUE THREAD

**Issue by miguelriemoliveira** [lardemua/AtlasCarCalibration#36]

**Comment by miguelriemoliveira** [comment]
FYI @afonsocastro For now just testing with the current cost function to see if it makes sense ...

**Comment by miguelriemoliveira** [comment]
Hi @afonsocastro ,

I made the following test: **built a json file with a bad first guess** [images are not very nice but for that we have #50 ].

![Screenshot at 2019-11-13 02-58-16](https://user-images.githubusercontent.com/11350998/68729345-a1d2df80-05c1-11ea-9bac-70589914e0ed.png)
![Screenshot at 2019-11-13 02-58-25](https://user-images.githubusercontent.com/11350998/68729348-a26b7600-05c1-11ea-9ec9-623e548f6b31.png)
![Screenshot at 2019-11-13 02-58-34](https://user-images.githubusercontent.com/11350998/68729349-a26b7600-05c1-11ea-9a56-b0c8f38828f7.png)
  
**then ran an optimization with all sensors and about 40 collections.** the goal was to see if the pose of the lidars converged to the real pose. Results are these:

![Screenshot at 2019-11-13 02-56-33](https://user-images.githubusercontent.com/11350998/68729395-caf37000-05c1-11ea-8f4d-54c4223b2671.png)
![Screenshot at 2019-11-13 02-56-44](https://user-images.githubusercontent.com/11350998/68729396-cb8c0680-05c1-11ea-875c-719fab9c25ce.png)

Bottom line, I am not sure this is working, the Lasers did not converge to the propper place did they?

**Comment by miguelriemoliveira** [comment]
... what else could we try? An optimization without cameras? not sure it would / should provide more information ... any ideas ?

**Comment by eupedrosa** [comment]
Maybe we should try an optimization without cameras and see what happens. The lasers may not align on the z-axis but they should align on xy-axis.

**Comment by miguelriemoliveira** [comment]
Try out new orthogonal residuals for all the points 

![image](https://user-images.githubusercontent.com/11350998/69805233-2899d600-11d8-11ea-8155-9693fa73274b.png)

**Comment by miguelriemoliveira** [comment]
Example of launchig the code

`test/sensor_pose_json_v2/main_ros_visualization.py -json ~/datasets/calibration_26nov2019/data_collected.json -cradius .5 -csize 0.101 -cnumx 9 -cnumy 6 -vo -si -csf "lambda x: not str(x) in ['5', '6', '7', '12', '13', '14', '15']  and int(x)<7" -ssf "lambda name: name=='top_left_camera' or name=='right_laser'"`

**Comment by afonsocastro** [comment]
Done
